connectors/kafka/src/main/java/io/camunda/connector/kafka/inbound/KafkaConnectorConsumer.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.kafka.inbound;
+
+import static io.camunda.connector.kafka.inbound.KafkaPropertyTransformer.convertConsumerRecordToKafkaInboundMessage;
+import static io.camunda.connector.kafka.inbound.KafkaPropertyTransformer.getKafkaProperties;
+import static io.camunda.connector.kafka.inbound.KafkaPropertyTransformer.getOffsets;
+
+import io.camunda.connector.api.inbound.InboundConnectorContext;
+import io.camunda.connector.api.inbound.InboundConnectorResult;
+import io.camunda.connector.impl.ConnectorInputException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KafkaConnectorConsumer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaConnectorConsumer.class);
+
+  private InboundConnectorContext context;
+
+  public CompletableFuture<?> future;
+
+  Consumer<String, String> consumer;
+
+  boolean shouldLoop = true;
+
+  private KafkaConnectorProperties elementProps;
+
+  private final Function<Properties, Consumer<String, String>> consumerCreatorFunction;
+
+  public KafkaConnectorConsumer(
+      final Function<Properties, Consumer<String, String>> consumerCreatorFunction,
+      InboundConnectorContext connectorContext,
+      KafkaConnectorProperties elementProps) {
+    this.consumerCreatorFunction = consumerCreatorFunction;
+    this.context = connectorContext;
+    this.elementProps = elementProps;
+  }
+
+  public void startConsumer() {
+    this.consumer =
+        createConsumer(
+            this.consumerCreatorFunction,
+            getKafkaProperties(elementProps, this.context),
+            elementProps,
+            getOffsets(elementProps.getOffsets()));
+    LOG.info("Kafka inbound connector initialized");
+    this.future =
+        CompletableFuture.supplyAsync(
+            () -> {
+              try {
+                while (shouldLoop) {
+                  ConsumerRecords<String, String> records =
+                      this.consumer.poll(Duration.ofMillis(500));
```

Comment:
@igpetrov good idea, we can make this configurable, although we have to be careful. According to the documentation this poll time is "The time, in milliseconds, spent waiting in poll if data is not available in the buffer." It can block the thread for this amount of time, which might make it hard to deactivate the consumer. Maybe make it configurable as a next iteration? We also have to find the right default value. WDYT?

Reply:
I would prefer to make it configurable in the next iteration with default preset value of 500 ms and customer is free to change if they need. Otherwise, sounds good to me